### PR TITLE
Add Google Sign-Out functionality to sample app

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Add Google Sign-Out functionality to sample app

#### WHY

In order to allow to sign out so the sign-in flow can be tested multiples times.

#### HOW

Add the logic into AuthGoogleSignOutScreen / AuthGoogleSignOutViewModel, display a success confirmation message when successfully or fail silently for now.

#### Checklist :clipboard:
- [N/A] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [N/A] Update metalava's signature text files

### CODE REVIEW

File: sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+) : ViewModel() {
+
+    private val _uiState = MutableStateFlow(GoogleSignOutScreenState.Idle)
+    public val uiState: StateFlow<GoogleSignOutScreenState> = _uiState.stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(stopTimeoutMillis = 5_000),
+        initialValue = GoogleSignOutScreenState.Idle
+    )
+
+    init {
```
Comment (yschimke) [inline sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutViewModel.kt:47]: I know this is sample code, but do we have clear guidance for when to use `init{}` vs a SideEffect?

```
    when (state) {
        AuthGoogleSignInScreenState.Idle -> {
            SideEffect {
                if (!executedOnce) {
                    executedOnce = true
                    viewModel.startAuthFlow()
                }
            }
        }
```

File: sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutScreen.kt [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                    modifier = Modifier.align(Alignment.CenterHorizontally),
+                    textAlign = TextAlign.Center,
+                    text = stringResource(id = R.string.google_sign_out_success_message)
+                )
+            }
+        }
+
+        GoogleSignOutScreenState.Failed -> {
+            SideEffect {
+                navController.popBackStack()
```
Comment (yschimke) [inline sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutScreen.kt:59]: Should we show the error somehow? Snackbar?

**Review by yschimke — APPROVED** [review]

File: sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutScreen.kt (hunk 2 above)
Comment (luizgrp) [reply sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutScreen.kt:59]: could do, chosen to leave it for now as per pr description - it's logging the exception in logcat in the VM

File: sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutViewModel.kt (hunk 1 above)
Comment (luizgrp) [reply sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutViewModel.kt:47]: Not sure about the guidance, I'd personally don't like to have a VM with side effects in its constructor. 

In this case the call to sign out could have been done in the caller screen, it wouldn't need a screen and vm just for this. The only reason I added like this was to have all the code contained in a separate class inside a "googlesignin" package, thus avoiding the caller class `AuthMenuScreen` to become a god class.

Any strong opinions about this? I will iterate back on this, just needed this functionality for now and didn't spend much time in designing it.

Comment (yschimke) [reply sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutViewModel.kt:47]: No, mainly observing a mix in these recent PRs.

Comment (luizgrp) [reply sample/src/main/java/com/google/android/horologist/auth/googlesignin/GoogleSignOutViewModel.kt:47]: cool, will probably remove this init once improving the sign out and address the failure case from the comment below

**Pull request merged**